### DISCUSSION TITLE
TestFlight: Cannot add internal group to a build

### DISCUSSION THREAD

**Discussion by VIadislavJevremovic** [codemagic-ci-cd/codemagic-docs#2488] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

iOS (Swift, Objective-C)

### Steps to reproduce

I used the following snippet for TestFlight distribution in codemagic.yaml (this is the only troublesome part, the rest works quite well):

```
app_store_connect:
  auth: integration
  submit_to_testflight: true
  expire_build_submitted_for_review: true
  beta_groups:
    - Team-1
```

### Expected results

Build is assigned to a specific internal beta group on TestFlight.

### Actual results

The following error is encountered at the end of post-processing step (build is actually assigned to all the internal groups on TestFlight):

```
Adding build '902b5448-2874-4bef-9667-3cd385e5ec22' to the following beta groups: Team-1.
Failure: Failed to add a build '902b5448-2874-4bef-9667-3cd385e5ec22' to 'Team-1' beta group. Builds cannot be assigned to this internal group. - Cannot add internal group to a build.
```

(Beta group assignment does work well for Firebase App Distribution)

### Build id (optional)

66260fb568d504f180d0c1f0

**Comment by VIadislavJevremovic** [comment, marked as answer]
Solved: internal groups need to have Manual distribution type in TestFlight (not Automatic as it is by default).

**Comment by galvesribeiro** [comment]
Here is how it was fixed: https://github.com/fastlane/fastlane/issues/22051#issuecomment-2978738326
